src/Nethermind/Nethermind.Consensus.AuRa/AuRaContractGasLimitOverride.cs

```diff
@@ -0,0 +1,87 @@
+﻿//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.Collections.Generic;
+using Nethermind.Consensus.AuRa.Contracts;
+using Nethermind.Core;
+using Nethermind.Core.Caching;
+using Nethermind.Core.Crypto;
+using Nethermind.Crypto;
+using Nethermind.Dirichlet.Numerics;
+using Nethermind.Logging;
+
+namespace Nethermind.Consensus.AuRa
+{
+    public class AuRaContractGasLimitOverride : IGasLimitOverride
```

Comment:
why not AuRaGasLimitOverrideContract? if you want to have them next to each other on the list then mybe a directory?

Reply:
So we have 2 entities:
1. We have a contract
2. We have an override that uses a contract. 
This is the 2nd thing its not a contract.

Reply:
I see!